androidshared/src/main/java/org/odk/collect/androidshared/ui/FixMismatchThemeProblemsActivity.kt

```diff
@@ -0,0 +1,43 @@
+package org.odk.collect.androidshared.ui
+
+import android.os.Build
+import android.os.Bundle
+import android.webkit.WebView
+import androidx.appcompat.app.AppCompatActivity
+
+/**
+ * When the app's theme doesn't match the system, the first [WebView] created will cause the host
+ * Activity to be recreated with incorrect resource loading (the system ones instead of the custom
+ * set dark/light resources). This doesn't happen for subsequent [WebView] creations however.
+ *
+ * This Activity can be started at launch to be "sacrificed" so no other Activity has to deal
+ * with the problem.
+ *
+ * See [this issue](https://issuetracker.google.com/issues/37124582) for more details.
+ */
+class FixMismatchThemeProblemsActivity : AppCompatActivity() {
```

Comment:
Does it really need to be in an activity? Maybe it could be just a method in `ApplicationInitializer` instead?

Reply:
Ah great point! I'd assumed that we needed an Activity to "sacrifice", but it looks like if I stick the `WebView` initialisation in `ApplicationInitializer` it does fix the problems. Will rework this.